common/scala/src/main/scala/whisk/core/connector/MessageConsumer.scala

```diff
@@ -92,140 +92,151 @@ object MessageFeed {
+          // essentially marking the activation as having satisfied "at most once"
+          // semantics (this is the point at which the activation is considered started).
+          // If the commit fails, then messages peeked are peeked again on the next poll.
+          // While the commit is synchronous and will block until it completes, at steady
+          // state with enough buffering (i.e., maxPipelineDepth > maxPeek), the latency
+          // of the commit should be masked.
+          val records = consumer.peek(longPollDuration)
+          consumer.commit()
+          FillCompleted(records.toSeq)
         }
+      }.andThen {
+          case Failure(e: CommitFailedException) =>
+            logging.error(this, s"failed to commit $description consumer offset: $e")
+          case Failure(e: Throwable) =>
+            logging.error(this, s"exception while pulling new $description records: $e")
+        }
+        .recover {
+          case _ => FillCompleted(Seq.empty)
+        }
+        .pipeTo(self)
+    } else {
+      logging.error(this, s"dropping fill request until $description feed is drained")
     }
+  }
 
-    /** Send as many messages as possible to the handler. */
-    @tailrec
-    private def sendOutstandingMessages(): Unit = {
-        val occupancy = outstandingMessages.size
-        if (occupancy > 0 && handlerCapacity > 0) {
-            val (topic, partition, offset, bytes) = outstandingMessages.dequeue()
+  /** Send as many messages as possible to the handler. */
+  @tailrec
+  private def sendOutstandingMessages(): Unit = {
+    val occupancy = outstandingMessages.size
+    if (occupancy > 0 && handlerCapacity > 0) {
+      val (topic, partition, offset, bytes) = outstandingMessages.dequeue()
 
-            if (logHandoff) logging.info(this, s"processing $topic[$partition][$offset] ($occupancy/$handlerCapacity)")
-            handler(bytes)
-            handlerCapacity -= 1
+      if (logHandoff)
+        logging.info(this, s"processing $topic[$partition][$offset] ($occupancy/$handlerCapacity)")
+      handler(bytes)
+      handlerCapacity -= 1
 
-            sendOutstandingMessages()
-        }
+      sendOutstandingMessages()
     }
-
-    private def shouldFillQueue(): Boolean = {
-        val occupancy = outstandingMessages.size
-        if (occupancy <= pipelineFillThreshold) {
-            logging.debug(this, s"$description pipeline has capacity: $occupancy <= $pipelineFillThreshold ($handlerCapacity)")
-            true
-        } else {
-            logging.debug(this, s"$description pipeline must drain: $occupancy > $pipelineFillThreshold")
-            false
-        }
+  }
+
+  private def shouldFillQueue(): Boolean = {
+    val occupancy = outstandingMessages.size
+    if (occupancy <= pipelineFillThreshold) {
+      logging.debug(
+        this,
+        s"$description pipeline has capacity: $occupancy <= $pipelineFillThreshold ($handlerCapacity)")
```

Comment:
is it possible to wrap only the last arg, instead of causing all args to appear wrapped? like:
```
logging.debug(this, 
  s"$description pipeline has capacity: $occupancy <= $pipelineFillThreshold ($handlerCapacity)")
```